homeassistant/components/media_player/kodi.py

```diff
@@ -520,25 +564,31 @@ def supported_features(self):
         """Flag media player features that are supported."""
         supported_features = SUPPORT_KODI
 
-        if self._turn_off_action in TURN_OFF_ACTION:
+        if self._turn_on_action is not None:
+            supported_features |= SUPPORT_TURN_ON
+
+        if self._turn_off_action is not None:
             supported_features |= SUPPORT_TURN_OFF
 
         return supported_features
 
     @cmd
     @asyncio.coroutine
+    def async_turn_on(self):
+        """Execute turn_on_action to turn on media player."""
+        if self._turn_on_action is not None:
+            yield from self._turn_on_action.async_run()
```

Comment:
Ups, no, I can't. Now I need the `async_update_ha_state` and `async_on_quit` entity methods to create the scripts... I'll simply change the template and move out the deprecation conversion